pjlib/include/pj/config.h

```diff
@@ -1024,6 +1024,18 @@
 #  define PJ_SSL_SOCK_MAX_CURVES   32
 #endif
 
+/**
+ * Use OpenSSL thread locking callback. This is only applicable for OpenSSL
+ * version prior to 1.1.0
+ *
+ * Default: 1 (enabled)
+ */
+#ifndef PJ_SSL_SOCK_OSSL_USE_THREAD_CB
+#   define PJ_SSL_SOCK_OSSL_USE_THREAD_CB   1
+#else
+#   define PJ_SSL_SOCK_OSSL_USE_THREAD_CB   0
+#endif
```

Comment:
hmm, why not enable by default if version < 1.1.0, e.g:
```
#ifndef PJ_SSL_SOCK_OSSL_USE_THREAD_CB
#  if OPENSSL_VERSION_NUMBER < 0x10100000L
#    define PJ_SSL_SOCK_OSSL_USE_THREAD_CB    1
#  else
#    define PJ_SSL_SOCK_OSSL_USE_THREAD_CB    0
#  endif
#endif
```

Reply:
version number check shouldn't be on config.h since it needed to include opensslv.h (openssl header) and other version checking (existing ones) is done on ssl_sock_ossl.c

Reply:
ah okay. just a note, if openssl >= 1.1.0 supports this locking mech, we may need to include opensslv.h here, should be fine if openssl is enabled.